### DISCUSSION TITLE
Nx Repository Splitting

### DISCUSSION THREAD

**Discussion by kauppfbi** [nrwl/nx#11781] [General]
Hi 👋

I would like to discuss possible approaches which support a `split` of a Nx Monorepository. I know it sounds wrong and is contrary to the idea of monorepos, but here is a scenario, where it might make sense. 

Imagine you have a big mono repository based on Nx with multiple projects, applications and libraries. However this mono repository is scoped to a certain organisation within a company. The scope should remain fixed. The ground work for all the applications are many shared libraries, primarily ui components, which are also distributed as npm packages.

This UI component library is of interest for other departments or even the open source world and now we would like to encourage the collaboration with other parties (company internal or even in the open source world). But here are the constraints: 

- Integrating other departments projects is not desired/possible/...
- Opening the workspace on Github public is also not possible as we have application code worth protecting in the repository as well
- Just extracting the ui libraries to a new repository is also not desired --> existing applications would loose their great benefits of continuous integration, maintenance effort would increase, ... 
I can barely remember that @santoshyadavdev once posted some similar idea on twitter, hope this is right?

I recently came across bit.dev, which is - from my understanding - focusing on the distributed collaborative development of components. Ultimately you can always import components for further development, export new versions and ejecting them again. However, I personally find it not mature, especially with Angular. On the other site it would mean a replacement for Nx and Storybook, which I don't like tbh. 

Another option would be to work with git-submodules, but I am not a big fan of the idea as well. I definitely see the complexity increasing and I even would not be sure, if it is working anyhow.

Do you have any ideas? Are we still on the right track?

I appreciate all your ideas & thoughts.  🙏

**Comment by santoshyadavdev** [comment]
Yeah me and @SerkanSipahi tried to do something on the same lines, we were not that successful and had to scrap the solution which we tried

**Reply by kauppfbi** [reply]
What was it, that you had in mind?
